Add PostgreSQL Enhanced addon for Gramps 6.0

glamberson opened the pull request:
## Description

This PR adds the PostgreSQL Enhanced addon, which provides an advanced PostgreSQL database backend for Gramps.

## Key Features

- **Modern psycopg (v3)**: Uses the latest PostgreSQL adapter (not the deprecated psycopg2)
- **Dual storage format**: Maintains both pickle blobs (for Gramps compatibility) and JSONB (for advanced queries)
- **Two database modes**: 
  - Monolithic mode: All trees in one database with table prefixes
  - Separate mode: Each tree gets its own database
- **Advanced queries**: Recursive CTEs for relationship finding, full-text search capabilities
- **Performance**: Tested with 100,000+ person databases, 3-10x faster than SQLite
- **Connection pooling**: Better resource management for concurrent access

## Testing

- Extensively tested with Gramps 6.0.3
- Successfully imported and managed databases with 100,000+ persons
- Performance benchmarks documented in repository
- Full compatibility with all Gramps tools and reports verified

## Requirements

- PostgreSQL 15 or higher
- Python 3.9+
- psycopg>=3 (installed via `pip install 'psycopg[binary]'`)

## Status

This is marked as **experimental** (version 0.1.0) and targeted at developers and advanced users who need high-performance database capabilities. While thoroughly tested, it should be considered experimental until more users have tested it in production.

## Documentation

Full documentation available at: https://github.com/glamberson/gramps-postgresql-enhanced

## Author

Greg Lamberson (greg@aigenealogyinsights.com)

lamco-office committed:
Add PostgreSQL Enhanced addon (experimental)

- EXPERIMENTAL: Advanced PostgreSQL backend using psycopg3
- Dual storage with pickle blobs and JSONB
- Advanced query capabilities with recursive CTEs
- Connection pooling for better performance
- Tested with 100,000+ person databases
- Full Gramps compatibility maintained
- Version 0.1.0 - Developer audience

lamco-office committed:
Fix maintainer email address

Changed from lamberson@yahoo.com to greg@aigenealogyinsights.com

GaryGriffin commented:
The publish step failed. You have 2 errors in the template.pot file:

PostgreSQLEnhanced/po/template.pot:128: duplicate message definition...
PostgreSQLEnhanced/po/template.pot:95: ...this is the location of the first definition
PostgreSQLEnhanced/po/template.pot:162: duplicate message definition...
PostgreSQLEnhanced/po/template.pot:34: ...this is the location of the first definition

glamberson commented:
Thank you for catching those duplicate message definitions\! 

I've fixed the template.pot file by removing the duplicates:
- Removed duplicate 'Starting upgrade...' message (was at lines 95 and 128)  
- Removed duplicate 'psycopg3 is required...' message (was at lines 34 and 162)

The changes have been pushed in commit 3d00671f. The publish step should now pass without errors.

GaryGriffin commented:
I merged your branch in https://github.com/gramps-project/addons-source/commit/4b9a47cf364ee96471d29ac8d15955e4a39c9c9d  before you made this last commit (for template.pot). You need to create a new PR for the template.pot change.

glamberson commented:
Thanks @GaryGriffin\! I've created PR #751 with just the template.pot fix: https://github.com/gramps-project/addons-source/pull/751
